app/src/main/kotlin/com/wire/android/ui/home/conversations/details/GroupConversationDetailsViewModel.kt

```diff
@@ -5,88 +5,168 @@ import androidx.compose.runtime.mutableStateOf
     savedStateHandle: SavedStateHandle,
     qualifiedIdMapper: QualifiedIdMapper
-) : GroupConversationParticipantsViewModel(savedStateHandle, navigationManager, observeConversationMembers, qualifiedIdMapper) {
+) : GroupConversationParticipantsViewModel(
+    savedStateHandle,
+    navigationManager,
+    observeConversationMembers,
+    qualifiedIdMapper
+) {
 
     override val maxNumberOfItems: Int get() = MAX_NUMBER_OF_PARTICIPANTS
 
     private val conversationId: QualifiedID = qualifiedIdMapper.fromStringToQualifiedID(
         savedStateHandle.get<String>(EXTRA_CONVERSATION_ID)!!
     )
 
-    var groupOptionsState: GroupConversationOptionsState by mutableStateOf(GroupConversationOptionsState())
+    var groupOptionsState: GroupConversationOptionsState by mutableStateOf(GroupConversationOptionsState(conversationId))
+    var requestInProgress: Boolean by mutableStateOf(false)
 
     init {
         observeConversationDetails()
     }
 
     private fun observeConversationDetails() {
         viewModelScope.launch {
-            // TODO(QOL): refactor to one usecase that return group info and members
-            observeConversationMembers(conversationId)
-                .map { it.isSelfAnAdmin }
-                .distinctUntilChanged()
-                .also { isSelfAdminFlow ->
-                    val conversationDetailsFlow = observeConversationDetails(conversationId)
-                        .filterIsInstance<ObserveConversationDetailsUseCase.Result.Success>() // TODO handle StorageFailure
-                        .map { it.conversationDetails }
-                    combine(conversationDetailsFlow, isSelfAdminFlow, getSelfTeam()) { conversationDetails, isSelfAnAdmin, selfTeam ->
-                        Triple(
-                            conversationDetails,
-                            isSelfAnAdmin,
-                            selfTeam?.id != null && selfTeam.id == conversationDetails.conversation.teamId?.value
-                        )
-                    }.collect { (conversationDetails, isSelfAnAdmin, isSelfInOwnerTeam) ->
-                        with(conversationDetails) {
-                            if (this is ConversationDetails.Group) {
-                                updateState(
-                                    groupOptionsState.copy(
-                                        groupName = conversation.name.orEmpty(),
-                                        isUpdatingAllowed = isSelfAnAdmin,
-                                        areAccessOptionsAvailable = conversation.isTeamGroup(),
-                                        isGuestAllowed = (conversation.isGuestAllowed() || conversation.isNonTeamMemberAllowed()),
-                                        isServicesAllowed = conversation.isServicesAllowed(),
-                                        isUpdatingGuestAllowed = isSelfAnAdmin && isSelfInOwnerTeam
-                                    )
+            observerSelfUser().first()
```

Comment:
can we use collect here instead of first(), 
it's better to update the usecase to not return flow if we don't need to observe the changes